common-tooling/action.yml

```diff
@@ -0,0 +1,302 @@
+    default: 'true'
+  # Python
+  python-version:
+    description: "Version range or exact version of Python or PyPy to use, using SemVer's version range syntax. Reads from .python-version if unset."
+    default: "3.11"
+  python-version-file:
+    description: "File containing the Python version to use. Example: .python-version"
+  python-cache:
+    description: "Used to specify a package manager for caching in the default directory. Supported values: pip, pipenv, poetry."
+    required: false
+  python-architecture:
+    description: "The target architecture (x86, x64) of the Python or PyPy interpreter."
+  python-check-latest:
+    description: "Set this option if you want the action to check for the latest available version that satisfies the version spec."
+    default: "false"
+  python-token:
+    description: "The token used to authenticate when fetching Python distributions from https://github.com/actions/python-versions. When running this action on github.com, the default value is sufficient. When running on GHES, you can pass a personal access token for github.com if you are experiencing rate limiting."
+    default: ${{ github.server_url == 'https://github.com' && github.token || '' }}
+  python-cache-dependency-path:
+    description: "Used to specify the path to dependency files. Supports wildcards or a list of file names for caching multiple dependencies."
+  python-update-environment:
+    description: "Set this option if you want the action to update environment variables."
+    default: 'true'
+  python-allow-prereleases:
+    description: "When 'true', a version range passed to 'python-version' input will match prerelease versions if no GA versions are found. Only 'x.y' version range is supported for CPython."
+    default: "false"
+  python-enabled:
+    description: 'Whether to install python or not'
+    default: 'true'
+  # misc
+  overwrite:
+    description: 'Defines whether on hosted runners the present version should be overwritten'
+    default: 'true'
+  secrets:
+    required: false
+    default: ''
+    description: toJSON passed GitHub secrets
+  berlin-timezone:
+    description: 'Whether to keep the runner at UTC or set Berlin timezone'
+    default: 'true'
+
+
+runs:
+  using: composite
+  steps:
+  # figure out what's installed and potentially skip
+  - name: Assess System
+    shell: bash
+    run: |
+      if ! command -v node &> /dev/null
+      then
+          echo "node_missing=true" >> "$GITHUB_ENV"
+      fi
+
+      if ! command -v docker buildx &> /dev/null
+      then
+          echo "buildx_missing=true" >> "$GITHUB_ENV"
+      fi
+
+      if ! command -v python &> /dev/null
+      then
+          echo "python_missing=true" >> "$GITHUB_ENV"
+      fi
+
+      if ! command -v yarn &> /dev/null
+      then
+          echo "yarn_missing=true" >> "$GITHUB_ENV"
+      fi
+
+      if ! command -v java &> /dev/null
+      then
+          echo "java_missing=true" >> "$GITHUB_ENV"
+      fi
```

Comment:
maybe add something like `echo $GITHUB_ENV` here to see what tools are missing - hard to see that info in the GHA logs.

Reply:
added